### ISSUE TITLE
Incorrect warnings received while scanning dotnet core sdk 8.0 cbl_mariner image 

### ISSUE THREAD

**Issue by vishwesh-sharma** [anchore/syft#2525]
**What happened**:
Warnings received while scanning docker image mcr.microsoft.com/dotnet/sdk:8.0-cbl-mariner2.0-arm64v8 for json.sha256 files
**What you expected to happen**:
Syft should only scan for json for sbom an not json.sha256 files as they would not contain any sbom content
**Steps to reproduce the issue**:
syft mcr.microsoft.com/dotnet/sdk:8.0-cbl-mariner2.0-arm64v8 
**Anything else we need to know?**:
trivy is able to parse the image without any issues and generate a proper sbom
**Environment**:
- Output of `syft version`: 
- Application: syft
Version:    0.101.1
BuildDate:  2024-01-19T21:51:55Z
GitCommit:  Homebrew
GitDescription: [not provided]
Platform:   darwin/arm64
GoVersion:  go1.21.6
Compiler:   gc

- OS (e.g: `cat /etc/os-release` or similar): Darwin LY934P37T2 23.2.0 Darwin Kernel Version 23.2.0: Wed Nov 15 21:55:06 PST 2023; root:xnu-10002.61.3~2/RELEASE_ARM64_T6020 arm64

**Comment by tgerla** [comment]
Hi @vishwesh-sharma, I've tried scanning that image and I see the warnings but they do not seem to prevent an SBOM from being generated. Are you seeing an actual failure, or just warnings? I am only seeing warnings.

You can also disable the SBOM cataloger with this option on the command line:

`--catalogers "-sbom-cataloger"`

Hope this helps!

**Comment by vishweshsharma-1** [comment]
Issue has been addressed in 0.103.1 , suggested option 
`--catalogers "-sbom-cataloger"` 
does not work as it is deprecated, documentation needs to be updated to reflect usage of --select-catalogers
`syft mcr.microsoft.com/dotnet/sdk:8.0-cbl-mariner2.0-arm64v8  --select-catalogers "-sbom-cataloger"`

**Comment by willmurphyscode** [comment]
In Syft 103.1, the sbom cataloger is disabled by default, running Syft on the image without `--select-catalogers` produces no warning:

```
❯ syft mcr.microsoft.com/dotnet/sdk:8.0-cbl-mariner2.0-arm64v8 -o json=/tmp/foo.json
 ✔ Loaded image                                                                                                                                     mcr.microsoft.com/dotnet/sdk:8.0-cbl-mariner2.0-arm64v8
 ✔ Parsed image                                                                                                                     sha256:8249b04474832cac13b88a4125340ced3ec97916e230192bdf5c8a9c417abc0a
 ✔ Cataloged contents                                                                                                                      3a5fc6da133508de4e35e2271b43f9dbdce1eca5eae07c084ffad545e7dd515b
   ├── ✔ Packages                        [3,996 packages]
   ├── ✔ File digests                    [10,874 files]
   └── ✔ File metadata                   [10,874 locations]
```

https://github.com/anchore/syft?tab=readme-ov-file#package-cataloger-selection has been updated to references `--select-catalogers` instead of `--catalogers`.

I'm going to close this as fixed by #2527, but please let us know if we're missing something.
